### PR TITLE
ci: run test for PR

### PR DESCRIPTION
**Pull request by lwshang**

### CODE REVIEW

**Comment by krpeacock** [comment]
@lwshang does the pull_request entry not need anything under it?

File: .github/workflows/examples.yml [hunk 1]
```diff
@@ -1,7 +1,11 @@
 
 name: Examples
 
-on: [push]
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```
Comment (ericswanson-dfinity) [inline .github/workflows/examples.yml:8]:
```suggestion
  pull_request:
    types:
      - opened
      - reopened
      - edited
      - synchronize
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -1,6 +1,10 @@
 name: Tests
 
-on: [push]
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```
Comment (ericswanson-dfinity) [inline .github/workflows/test.yml:7]:
```suggestion
  pull_request:
    types:
      - opened
      - reopened
      - edited
      - synchronize

```

**Comment by ericswanson-dfinity** [comment]
We have a few different forms across our repos, but here is a suggestion for a list of pull request events.

**Comment by lwshang** [comment]
[GitHub doc](https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#pull_request) has this note:
> By default, a workflow only runs when a pull_request's activity type is opened, synchronize, or reopened. To trigger workflows for more activity types, use the types keyword.

IMO, this is sufficient for "test" and "examples".

For "check PR title", it includes an extra "edited" activity type. I believe it was intended for trigger CI when edited PR title.

**Comment by ericswanson-dfinity** [comment]
@lwshang Thanks for checking this.  I agree that the default behavior for `pull_request:` without specifying types is correct.

**Review by ericswanson-dfinity — APPROVED** [review]

**Pull request merged**